src/main/scala/reporting/Converter.scala

```diff
@@ -0,0 +1,503 @@
+package viper.silicon.reporting
+
+import scala.util.{Try, Success}
+import viper.silver.verifier.{Model, ModelEntry, ValueEntry, ConstantEntry, ApplicationEntry, MapEntry}
+import viper.silver.ast
+import viper.silicon.interfaces.state.Chunk
+import viper.silicon.resources.{FieldID, PredicateID}
+import viper.silicon.state.{Store, State, BasicChunk}
+import viper.silicon.state.terms._
+import viper.silicon.decider.TermToSMTLib2Converter
+
+//Classes for extracted Model Entries
+case class ExtractedModel(entries: Map[String, ExtractedModelEntry]) {
+  override lazy val toString: String = {
+    entries.map(x => s"${x._1} <- ${x._2.toString}").mkString("\n")
```

Comment:
Iteration order of `Map` is probably nondeterministic, resulting in a nondeterministic `toString` (across multiple runs), which may be confusing. You could use a deterministic alternative map (see https://github.com/viperproject/silicon/blob/master/src/main/scala/package.scala#L17), or sort by keys before creating the string.

Reply:
https://github.com/viperproject/silicon/blob/0c495c0fc14f2b22f6710167ab4b86a076f46790/src/main/scala/package.scala#L17-L18

I think the keyword `Map` here refers to the above type definition. In this case the order should be deterministic, right? At least that was my understanding when I coded this, because the implementation using this in Prusti also relies on these maps being deterministic and in my use-cases they were.

Reply:
IntelliJ will tell you, if you hover over the type or go to its definition, if I'm not mistaken. If so, you can indeed ignore my comment.

Reply:
Yeah I was wrong, the definition was taken from `scala.Predef`. But it is fixed now.